cdap-common/src/test/java/co/cask/cdap/common/ssh/SSHSessionTest.java

```diff
@@ -169,6 +189,120 @@ public void testSsh() throws Exception {
     }
   }
 
+  @Test
+  public void testLocalPortForwarding() throws Exception {
+    // Starts an echo server for testing the port forwarding
+    EchoServer echoServer = new EchoServer();
+
+    echoServer.startAndWait();
+    try {
+      // Creates the DataConsumer for receiving data and validating the lifecycle
+      StringBuilder received = new StringBuilder();
+      AtomicBoolean finished = new AtomicBoolean();
+
+      PortForwarding.DataConsumer dataConsumer = new PortForwarding.DataConsumer() {
+        private final List<String> messages = new ArrayList<>();
+
+        @Override
+        public void received(ByteBuffer buffer) {
+          messages.add(StandardCharsets.UTF_8.decode(buffer).toString());
+        }
+
+        @Override
+        public synchronized void flushed() {
+          messages.forEach(received::append);
+        }
+
+        @Override
+        public void finished() {
+          finished.set(true);
+        }
+      };
+
+
+      SSHConfig sshConfig = getSSHConfig();
+
+      // Creates a SSH session.
+      try (SSHSession session = new DefaultSSHSession(sshConfig)) {
+        InetSocketAddress bindAddress = echoServer.getBindAddress();
+
+        // Creates local port forward and send data to the echo server through that forwarding channel
+        try (PortForwarding portForwarding = session.createLocalPortForward(bindAddress.getHostName(),
+                                                                            bindAddress.getPort(),
+                                                                            12345, dataConsumer)) {
```

Comment:
can this fail if port 12345 happens to be in use?

Reply:
No it won't.